### DISCUSSION TITLE
Accelerate shipping with GitHub projects 🏎

### DISCUSSION THREAD

**Discussion by AndreaGriffiths11** [community/community#30707] [Other Feature Feedback, Questions, & Ideas]
Organizations today are under pressure to do more with less. That's why productivity is such an important issue. And that's where project planning comes in.

Project planning is a critical tool for organizations of all sizes. It can help you ship products faster, increase efficiency, and improve communication among team members. In this demo, Senior Software Engineer Jed Verity will show you how to use GitHub Projects to boost productivity in your organization. You'll learn how to create and manage projects, set deadlines, and track progress. You'll also learn how to use project planning to resolve conflicts and make better decisions about priorities. By the end of this talk, you'll have a better understanding of how project planning with GitHub Projects can help your organization succeed.

We'll answer questions post event in this discussion, feel free to join: https://www.youtube.com/watch?v=a40ecigNIdw

**Comment by lfrancke** [comment]
Thank you for the presentation. We just tried to migrate from ZenHub to Github projects and ran into a few snags (which I'd be happy to list but will obviously be highly subjective).
Is there anything you can and/or are willing to share about the next few weeks/months that's more detailed than the roadmap? (https://github.com/orgs/github/projects/4247/views/7)

**Reply by lfrancke** [reply]
You haven't asked but maybe the feedback is helpful anyway so I'll just list it here. We're trying to implement a Kanban workflow using Projects and it's not easy.
A few things that are missing for us/that we'd like to see:

* Better/easier workflows
  * A way to add new issues from repositories automatically to a project (for triaging)
  * A way to automatically add/move issues to another project when it enters a column (e.g. an item in a "Done" column in a  "Design" project should be added to the "Development" project)
* A description/label per column that can be shown at the top
* Make it possible to link a PR to an issue in a different repository
- Show more details about "nested" PRs (e.g. reviewers) so we only have to add one of the two to the board (PR and/or issue) or at least be able to move them around quickly together
- WIP limits (at least visualize them)
- Mark an issue as blocked by another one including a proper link
- Option to hide empty columns in board views
- Allow "OR" filters or extend the 256 char limit
- Some way to tackle hierarchical tasks (Epics -> Stories -> Tasks)
- Swimlanes within a board view (e.g. an "Expedite" lane)

**Reply by glortho** [reply]
Thank you very much for this feedback @lfrancke! I'm sharing it in our internal team discussion forum. Some of these are definitely coming before long -- for example [hierarchy](https://github.com/github/roadmap/issues/284). 

I'm sorry to hear of your troubles migrating from ZenHub. Could you describe your snags in a bit more detail? I'll see if we can help. It might be best to open a targeted discussion [here](https://github.com/orgs/community/discussions/categories/issues).

**Reply by lfrancke** [reply]
Done: https://github.com/orgs/community/discussions/31227

**Comment by erinmez** [comment]
Thanks a lot for the presentation, especially the approach with multiple stati was interesting. 

We would like to move away from another kanban tool and would need WIP limits and non-removable note stickers in or above the columns as guides for team members when a ticket is deemed worthy moving into the next column. 
Parent child linkage would be nice too -> when at least one child ticket is In Progress, the parent moves into "In Progress". When all child tickets are "Done", the parent moves into "Done" etc.  

Any plans for those features?

**Reply by lfrancke** [reply]
Because we're trying the same (see my answer/question) I'd be very interested to see in what you came up with and if it works for you.
We've been part of the beta for months and have tried Github projects every few weeks again but there are a lot of small issues with it that prevent us from really using it heavily yet. The points you note we also face for example.

**Reply by glortho** [reply]
Ah I totally feel your pain on these @erinmez. I'm adding this feedback also to our internal discussion board. For now I can only share what's in the [public roadmap](https://github.com/orgs/github/projects/4247/views/7), but we're working hard to get these kinds of improvements out the door ASAP!

**Comment by krisu5** [comment]
I'm disappointed seeing that "Projects" is more and more just "issues extended" type of tool and **not able to make proper markdown note cards**. Because markdown note cards are used by many people[^1][^2][^3][^4][^5][^6][^7][^8][^9][^10] <sup>(and more)</sup> and now because of removal of the function, new projects is actually more limited then classic projects.

And since the "new projects" doesn't have beta label anymore, I expect classic projects sunset in near future.

I personally have **only used notes** in the classic projects, almost never listed issues with it and I found it reduntant for my workflow when ever I did so.

I tried the migration with [my repo's classic project](https://github.com/aegis-icons/aegis-icons/projects/1) and found out what a downgrade it was for my usage case.

All the cards become "drafts" (whatever those supposed to be, I don't know and don't care TBH) and now there's no way to interact or **even see the body of the cards** (yes, I know you can click to see the full info, but it still is massive downgrade compare how things were before).

At least **seeing the markdown body content and interacting it in the kanban board is must**, before I actually move from classic projects to new projects. **Doesn't matter if table mode doesn't have similar features**, I (and others) only care about kanban board.

[You promised markdown notes for new projects, but it silently got removed from roadmap.](https://github.com/community/community/discussions/4422#discussioncomment-2085342)

At this point, if classic projects gets removed with no new projects's improvements in related to markdown notes, I will move to some other kanban board tool.

*Sorry for the slightly annoyed tone of the comment, but I feel like left out as a user.*

[^1]: https://github.com/community/community/discussions/4422
[^2]: https://github.com/community/community/discussions/6604
[^3]: https://github.com/community/community/discussions/20180
[^4]: https://github.com/community/community/discussions/5795
[^5]: https://github.com/community/community/discussions/7378
[^6]: https://github.com/community/community/discussions/6721
[^7]: https://github.com/community/community/discussions/7544
[^8]: https://github.com/community/community/discussions/14651
[^9]: https://github.com/community/community/discussions/7098
[^10]: https://github.com/community/community/discussions/17680
